cypress/e2e/newCompanySpec.cy.js

```diff
@@ -0,0 +1,112 @@
+import { mockCompanies } from "../support/mockCompanies";
+
+describe("New Company page after logging in", () => {
+  let uniqueCompanyName = `Test Company ${Date.now()}`;
+
+  beforeEach(() => {
+    cy.visit("http://localhost:3000/");
+
+    cy.intercept('POST', 'http://localhost:3001/api/v1/users/2/companies', {
+      statusCode: 201,
+      body: {
+        id: 123,
+        name: uniqueCompanyName,
+        website: 'www.testcompany.com',
+        street_address: '123 Test St',
+        city: 'Test City',
+        state: 'CO',
+        zip_code: '12345',
+        notes: 'Test notes'
+      }
+    }).as('addCompany');
+    
+    cy.intercept("GET", "http://localhost:3001/api/v1/users/2/companies", {
+      statusCode: 200,
+      body: {
+        data: [
+          ...mockCompanies.data, 
```

Comment:
Good use of mocking (and a spread operator!), nice to get into the practice to save money and prevent API rate limits in future projects.